apps/cowswap-frontend/src/modules/swap/updaters/SmartSlippageUpdater.ts

```diff
@@ -42,12 +44,115 @@ export function SmartSlippageUpdater() {
 
       return response.slippageBps
     },
-    SWR_OPTIONS
+    SWR_OPTIONS,
   ).data
 
+  // TODO: remove v1
+  const tradeSizeSlippageBpsV1 = useSmartSlippageFromFeePercentage()
+  const tradeSizeSlippageBps = useSmartSlippageFromFeePercentageV2()
+
+  useEffect(() => {
+    // Add both slippage values, when present
+    const slippage = tradeSizeSlippageBps + (bffSlippageBps || 0)
+
+    setSmartSwapSlippage(slippage)
+  }, [bffSlippageBps, setSmartSwapSlippage, tradeSizeSlippageBps])
+
+  // TODO: remove before merging
   useEffect(() => {
-    setSmartSwapSlippage(typeof slippageBps === 'number' ? slippageBps : null)
-  }, [slippageBps, setSmartSwapSlippage])
+    console.log(`SmartSlippageUpdater`, {
+      granularSlippage: tradeSizeSlippageBpsV1,
+      fiftyPercentFeeSlippage: tradeSizeSlippageBps,
+      bffSlippageBps,
+    })
+  }, [tradeSizeSlippageBpsV1, tradeSizeSlippageBps])
 
   return null
 }
+
+const FEE_MULTIPLIER_FACTOR = new Fraction(15, 10) // 50% more fee, applied to the whole value => 150% => 15/10 in fraction
+const ONE = new Fraction(1)
+const ZERO = new Fraction(0)
+
+/**
+ * Calculates smart slippage in bps, based on quoted fee
+ *
+ * Apply a multiplying factor to the fee (e.g.: 50%), and from there calculate how much slippage would be needed
+ * for the limit price to take this much more fee.
+ * More relevant for small orders in relation to fee amount, negligent for larger orders.
+ */
+function useSmartSlippageFromFeePercentageV2(): number {
+  const { trade } = useDerivedSwapInfo() || {}
```

Comment:
I'm not sure about that, but if possible, would be better to replace `useDerivedSwapInfo` usage with `useReceiveAmountInfo`. `useDerivedSwapInfo` is a bit legacy.

Reply:
Also, `useDerivedSwapInfo` is a wrapper on top of `getQuoteAmountsAndCosts()` function from cow-sdk, so you could even move your fee logic in the SDK.